extensions/wikia/UserLogin/UserSignupSpecialController.class.php

```diff
@@ -122,82 +125,84 @@ public function signupForm() {
 		// process signup
 		$redirected = $this->request->getVal( 'redirected', '' );
 		if ( $this->wg->Request->wasPosted() && empty( $redirected ) ) {
+			$this->handleSignupFormSubmit();
+		} else {
+			$this->track( 'signup-start' );
```

Comment:
Quite a conversation!

So to start, why would I always need to call a static function with ```self```?  Its perfectly legal to call a static function with the ```$this``` instance variable.  If ```track``` was defined in a subclass of ```UserSignupSpecialController```, I'd end up running the parent version rather than the child.  If I specifically wanted the parent method, then ```self``` would be the way to go.  Without any other considerations, I'd say calling with this ```$this``` is preferable as it makes default, expected assumptions about inheritance.

The opposite, of course, is not true;  you don't want to call an instance method statically.  Even if it currently does not make use of ```$this``` it could, and PHP strict standards rightly scolds you for this.

To the next point, why did I make this static?  Copy and paste.  Generally I don't make things static unless there's a reason to do so.  I don't need to call this statically, nor do I see a reason why I would in the future.

So, I pulled the keyword and now its dynamic.  I also made it protected since there's no reason anyone outside this class and possible children who should be calling it.